UIdev/ObsMon/src/App.jsx

```diff
@@ -0,0 +1,264 @@
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/pstypes.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setPsTypes(data))
+      .catch(err => console.error("Failed to load psTypes:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/qtypes.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setQTypes(data))
+      .catch(err => console.error("Failed to load qTypes:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/ttypes.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setTTypes(data))
+      .catch(err => console.error("Failed to load tTypes:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/uvtypes.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setUvTypes(data))
+      .catch(err => console.error("Failed to load uvTypes:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/ozonesats.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setOzoneSatellites(data))
+      .catch(err => console.error("Failed to load ozoneSatellites:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/microwavesats.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setMicrowaveSatellites(data))
+      .catch(err => console.error("Failed to load microwaveSatellites:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/infraredsats.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setInfraredSatellites(data))
+      .catch(err => console.error("Failed to load infraredSatellites:", err));
+  }, []);
+
+  useEffect(() => {
+    fetch(withBase('data/geostationarysats.json'), { cache: 'no-store' })
+      .then(res => res.json())
+      .then(data => setGeostationarySatellites(data))
+      .catch(err => console.error("Failed to load geostationarySatellites:", err));
+  }, []);
+
+  const toggleSection = (name) => {
+    setOpenSection(openSection === name ? null : name);
+  };
+
+  const toggleSat = (name) => {
+    setOpenSat(openSat === name ? null : name);
+  };
+
+  const [cycleTime, setCycleTime] = useState(null);
+  const previousCycle = useRef(null);
+
+  // Load the current cycle and refresh periodically
+  useEffect(() => {
+    const fetchCycle = async () => {
+      try {
+        const res = await fetch(withBase('data/currentCycle.json'), { cache: 'no-store' });
+        const json = await res.json();
+        if (json.cycleTime && json.cycleTime !== previousCycle.cycleTime) {
+          setCycleTime(json.cycleTime);
+          previousCycle.cycleTime = json.cycleTime;
+        }
+      } catch (error) {
+        console.error('Failed to load current cycle:', error);
+      }
+    };
+
+    fetchCycle(); // Load on mount
+
+    const interval = setInterval(fetchCycle, 60000); // Poll every 60 seconds
```

Comment:
can you elaborate a bit more on this? Does this mean the page auto refreshes every minute?

Reply:
Honestly I don't have any idea about the load and what our server can support.  I went with 1 min because our "chatty" friend suggested it so I assumed it was fairly standard.  I think it's probably a good idea to stretch that refresh period to 5 min and set it to run only when in focus.